versioning/src/main/scala/org/typelevel/sbt/TypelevelVersioningPlugin.scala

```diff
@@ -37,6 +38,13 @@ object TypelevelVersioningPlugin extends AutoPlugin {
     lazy val tlUntaggedAreSnapshots =
       settingKey[Boolean](
         "If true, an untagged commit is given a snapshot version, e.g. 0.4-00218f9-SNAPSHOT. If false, it is given a release version, e.g. 0.4-00218f9. (default: true)")
+
+    lazy val tlCurrentRelease =
+      settingKey[Option[String]](
+        "The latest stable released version of your project, e.g. 0.2.0, 3.5.1. If applicable, this will be the version currently being released.")
+
+    lazy val tlCurrentPreRelease = settingKey[Option[String]](
+      "The latest pre-release (e.g. milestone, release candidate) of your project. If applicable, this will be the version currently being released.")
```

Comment:
Thanks, these are great descriptions! Bikeshed time, is "current release" and "current pre-release" the best terminology for these? Maybe "latest" is more appropriate? And "version" instead of "release"?

I guest that gets us to something like `tlLatestStableVersion` and `tlLatestPreReleaseVersion` maybe 🤔 bit of a mouthful.

Reply:
I have absolutely no opinion on this but happy to change it if you like 😃

Reply:
Not even a smidgeon of an opinion? 😂 I hate deciding these things. I can commit whatever I decide before I merge.

Reply:
Maybe I have a slight preference for `tlLatestStableRelease` and `tlLatestPreRelease` - explicit but reasonably concise.

Reply:
On the other hand, `currentRelease` isn't actually guaranteed to be a stable release:
> if the latest release is a pre-release (e.g., M or RC)
> and there are no stable releases it is bincompatible with,
> then for all effective purposes it is the current release

Reply:
Which comment specifically?

The idea is if you are on version `n` and you start prereleasing a binary-breaking version `n+1`, then it has effectively taken over that branch. If you want to keep version `n` alive then it would be on its own branch, where `currentRelease` would continue to be stable versions.

This was designed with http4s in mind: otherwise `currentRelease` on the `main` branch would always be 0.23.x and not the latest 1.0.0 milestone. Meanwhile, `currentRelease` on the `0.23.x` branch will always be a an 0.23.x version.


... but I'm totally open to better ideas here. Part of my hesitation with exposing these was that we need to be pretty solid on what we are exposing :)

Reply:
> Which comment specifically?

The one quoted in https://github.com/typelevel/sbt-typelevel/pull/333#discussion_r922100471

Your explanation makes sense though. So maybe the descriptions should be:

```suggestion
    lazy val tlCurrentRelease =
      settingKey[Option[String]](
        "The latest stable version of your project released from the current branch, e.g. 0.2.0, 3.5.1. If applicable, this will be the version currently being released.")

    lazy val tlCurrentPreRelease = settingKey[Option[String]](
      "The latest version of your project released from the current branch, whether stable or pre-release (e.g. milestone, release candidate). If applicable, this will be the version currently being released.")
```

Reply:
Yeah, in retrospect I can see calling it a "stable" release when there is no guarantee of that, is really not helpful :)

Reply:
I think the first one is maybe something like:
> The latest stable version of your project in the `tlBaseVersion` series, with the latest pre-release in this series used as a fallback if there is no stable release yet.

Reply:
It sounds like we need to make up a new word for the concept we're talking about here - `foo` such that https v1.0.0-M34 and v0.23.13 are foo versions but v0.23.14-RC1 would not be a foo version 🤔

Reply:
Maybe "current version" is the best we can do here. And then the description could be:

> The "current release" of your project from the current branch. Usually this will be the latest stable release. However, if your branch contains pre-releases that break binary compatibility with the latest stable release, the branch is considered a pre-release branch and the current release will therefore be the latest pre-release.

It's a bit wordy but I'm not sure how much more concise it's possible to be.

Reply:
Can I leave this with you to commit the final wording and naming? Happy to discuss ideas of course

Reply:
Yup, I can do that. Thanks for all your help, nice to bounce ideas! Much appreciated.

Reply:
> I think the first one is maybe something like:
> 
> > The latest stable version of your project in the `tlBaseVersion` series, with the latest pre-release in this series used as a fallback if there is no stable release yet.

Ah yeah sounds good, I only just saw this now.

Reply:
> The latest stable version of your project in the `tlBaseVersion` series, with the latest pre-release in this series used as a fallback if there is no stable release yet.

Actually I don't think that's correct? `tlBaseVersion` isn't used in the calculation, and this won't be the behaviour if (say) `tlBaseVersion` is 1.4 but there's a stable 1.3.0 release and a 1.4.0-M1 release, right?

Reply:
You are right, it's still not 100% correct.

> `tlBaseVersion` isn't used in the calculation

It is, albeit indirectly. `version` is used in the calculation, and `tlBaseVersion` is used to calculate `version`.

> `tlBaseVersion` is 1.4 but there's a stable 1.3.0 release and a 1.4.0-M1 release, right?

You are right about this. It uses the binary series and not the `x.y` series.